hack/errata-check.sh

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# RED HAT INTERNAL USE ONLY
+#
+# Purpose
+# -------
+# Which PRs are in the build currently attached to the errata for the next
+# release?
+#
+# The script will query Errata Tool for the active errata based on the
+# branch specified.  It will then look up the build attached and
+# determine the WMCO commit the build was built from.  Finally, it will list the
+# merge commits since the last WMCO release for that version.  The commit
+# associated with the build will be tagged with a shortened build NVR,
+# 'wmco-container-X.Y.Z-n'.
+#
+# Install Required Tools
+# ----------------------
+# dnf install curl jq gh
+#
+# Usage
+# -----
+# * VPN connectivity and valid kerberos ticket required
+# * Update your local clone and branches
+# * Run this script ./errata-check.sh <branch_name>
+#
+# Script Maintenance
+# ------------------
+# Several hashes for releases need to be maintained below.
+#
+# LAST_WMCO_HASH should be updated after a release for each version and after
+# branching day when the next release is no longer being fast forwarded.
+#
+# Finding the point where fast forwarding stopped for the last release:
+# NEXT_RELEASE="release-4.13"
+# LAST_RELEASE="release-4.12"
+# diff -u <(git rev-list --first-parent $NEXT_RELEASE) \
+#         <(git rev-list --first-parent $LAST_RELEASE) \
+#         | sed -ne 's/^ //p' | head -1
+
+# Set variables
+ERRATA_API_URL="https://errata.devel.redhat.com/api/v1/erratum"
+RED=$(tput setaf 1)
+GREEN=$(tput setaf 2)
+YELLOW=$(tput setaf 3)
+NORMAL=$(tput sgr0)
+BLUE=$(tput setaf 4)
+BRIGHT=$(tput bold)
+
+# Define functions
+function help() {
+  printf "${YELLOW}"
+  echo "Usage: errata-check.sh branch_name"
+  echo "List merge commits in the build attached to an active release errata"
+  echo ""
+  echo "Example:"
+  echo "# Check the build for the next WMCO v7 errata"
+  echo "errata-check.sh release-4.12"
+  printf "${NORMAL}"
+}
+
+function verify-args {
+  if [ $# -eq 0 ]
+  then
+    print-error "No branch provided"
+    help
+    exit 1
+  fi
+}
+
+function print-label {
+  local label=$1
+  local item=$2
+  printf "${BLUE}%-20s= ${NORMAL}%s\n" "$label" "$item"
+}
+
+function print-info {
+  local text=$1
+  printf "\n${BRIGHT}%s${NORMAL}\n" "$text"
+}
+
+function print-error {
+  local text=$1
+    printf "\n${RED}ERROR: %s${NORMAL}\n" "$text" 1>&2
+}
+
+function set-branch-vars {
```

Comment:
Instead of using a lookup table approach, I suggest determining the hash and version dynamically.

You can take in a branch as an argument, for example, `set-branch-vars release-4.13`.
```
local release_branch=$1
branch_prefix=$(semver=$(cut -d '-' -f 1 <<< "$input"))
semver=$(cut -d '-' -f 2 <<< "$input")
major_version=$(cut -d '.' -f 1 <<< "$semver")
minor_version=$(cut -d '.' -f 2 <<< "$semver")
previous_release_branch="4.$((minor_version-1))"
echo $previous_release_branch
```

You can then use the method you stated above to get the branching hash
```
# diff -u <(git rev-list --first-parent $NEXT_RELEASE) \
#         <(git rev-list --first-parent $LAST_RELEASE) \
#         | sed -ne 's/^ //p' | head -1 
```

The LAST_WMCO_VERSION can be found by looking at the release branch in question and searching `Makefile`:
```
$ git show release-4.12:Makefile |grep "WMCO_VERSION ?=
WMCO_VERSION ?= 7.0.0
```
The version can be captured directly using some bash regex magic, but I'm not sure what the best tool for that is .

Reply:
I had considered dynamically determining the last release, however my initial thought was to do that through another errata query for 'SHIPPED LIVE' errata.  That was going to be more expensive requiring more calls.  I like the suggestion of using git locally, however I'll need to think that through further.  When I do the `git show` mentioned above, I'm getting the "next" version for that branch.

```
$ git show release-4.12:Makefile |grep "WMCO_VERSION ?="
WMCO_VERSION ?= 7.1.0
```
Aravindh and I also discussed pushing tags to the repo for released versions, 7.0.0, 7.0.1, etc.  That would simplify the task of finding the commit used for the release without having to go through the errata.

`LAST_WMCO_VERSION` is only used in printing a description and `LAST_WMCO_HASH` is only used to limit the number of commits returned.  The vars are not really necessary to do most of the work in the script.

Reply:
If we do push tags I could do something like this:
```
$ git tag --merged release-4.12 | grep -E "^v" | sort --version-sort | tail -1
v7.0.1

$ git show-ref --tags "refs/tags/v7.0.1" | awk '{ print $1 }'                 
bc9473bf8c4200b0a016608e808122ef783570bc
```

If nothing is returned for the branch in the first step (no release for the branch), then it could drop back to running the diff to find the branching point.

Reply:
I am in favor of pushing tags, we have been thinking about it for a while now.

Reply:
+1

Reply:
Updated.  Hashes are no longer stored in the script now that release tags have been pushed.